### ISSUE TITLE
After upgrade to docker-ce 20.10.4 unable to start container with systemd

### ISSUE THREAD

**Issue by fansari** [docker/for-linux#1212]
* [x] This is a bug report
* [ ] This is a feature request
* [ ] I searched existing issues before opening this one

### Expected behavior

So far I was able to start containers like this. After upgrading from 20.10.3 to 20.10.4 this does not work anymore.

### Actual behavior
After running "systemctl start docker-mytest" I find the container in exit state.

### Steps to reproduce the behavior

Create a service file and put in to /etc/systemd/system/docker-mytest.service.

```
[Unit]
Description=test container
Requires=docker.service
After=docker.service

[Service]
Restart=always
KillMode=none
ExecStart=/usr/bin/docker start -a mytest
ExecStop=/usr/bin/docker stop -t 2 mytest

[Install]
WantedBy=multi-user.target
```

Reload systemd.

```
systemctl daemon-reload
```

Create files/run.sh.

```
#!/bin/sh

tail -f /dev/null
```

Make the file executable.

```
chmod 755 files/run.sh
```

Create a Dockerfile.

```
FROM alpine:latest
USER root
COPY files/run.sh /
CMD ["/run.sh"]
```

Pull an alpine image. Create an image an run it.

```
docker pull alpine:latest
docker build -t mytest .
docker run -d --name mytest myest
```
Now stop the container and try to start it with systemd with service file given above:

```
docker stop mytest
systemctl start docker-mytest
```

**Output of `docker version`:**

```
Client: Docker Engine - Community
 Version:           20.10.4
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        d3cb89e
 Built:             Thu Feb 25 07:04:27 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.4
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       363e9a8
  Built:            Thu Feb 25 07:02:50 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

**Output of `docker info`:**

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Build with BuildKit (Docker Inc., v0.5.1-docker)

Server:
 Containers: 3
  Running: 2
  Paused: 0
  Stopped: 1
 Images: 18
 Server Version: 20.10.4
 Storage Driver: overlay2
  Backing Filesystem: xfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc version: ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 4.18.0-277.el8.x86_64
 Operating System: CentOS Stream 8
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 3.648GiB
 Name: wes-zdb-001.vm.int.wesystems.cloud
 ID: NB5E:IZ3Z:XVCD:GRQN:2NAM:7VVJ:IZXQ:ZDOJ:EKDP:FQGZ:BVG4:5DQ5
 Docker Root Dir: /var/lib/database/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
```

**Additional environment details (AWS, VirtualBox, physical, etc.)**

VM running on Proxmox hypervisor.

**Comment by Qjammer** [comment]
We are having a similar problem with runnng a docker-compose command from a pre-commit hook, as such:
`docker-compose run --rm php-fpm php ./etc/git/hooks/pre-commit.php`
The command runs successfully when runnig from the console, but when it is run in the hook, the command hangs waiting for a process, as shown by strace:
`newfstatat(AT_FDCWD, "/usr/bin/docker", {st_mode=S_IFREG|0755, st_size=54360336, ...}, 0) = 0
getpid()                                = 62797
pipe2([20, 21], O_CLOEXEC)              = 0
clone(child_stack=NULL, flags=CLONE_CHILD_CLEARTID|CLONE_CHILD_SETTID|SIGCHLD, child_tidptr=0x7f66848d6a10) = 62808
close(21)                               = 0
read(20, "", 50000)                     = 0
close(20)                               = 0
wait4(62808, `

Running ps shows this process as such:
`my_username     62808  0.0  0.2 1498376 46436 pts/5   Sl+  15:07   0:00 /usr/bin/docker start --attach --interactive 807d53fbb165cae550260ed3374e00ce3e468c797a1ee44bfd7df32716358d7b`

**Comment by tomerd** [comment]
+1 seeing the same on CentOS 8

**Comment by Qjammer** [comment]
Fixed by https://github.com/docker/cli/pull/2960 in 20.10.5

**Comment by cpuguy83** [comment]
Indeed, this is fixed in 20.10.5.
Please let us know if you still have problems after upgrade (note specifically the CLI is the problem here so you must update that 20.10.5 engine is identical to 20.10.4).
Closing this for now.

**Comment by 76creates** [comment]
> We are having a similar problem with running a docker-compose command from a pre-commit hook, as such: `docker-compose run --rm php-fpm php ./etc/git/hooks/pre-commit.php` The command runs successfully when runnig from the console, but when it is run in the hook, the command hangs waiting for a process, as shown by strace:
> 
> ```
> newfstatat(AT_FDCWD, "/usr/bin/docker", {st_mode=S_IFREG|0755, st_size=54360336, ...}, 0) = 0
> getpid()                                = 62797
> pipe2([20, 21], O_CLOEXEC)              = 0
> clone(child_stack=NULL, flags=CLONE_CHILD_CLEARTID|CLONE_CHILD_SETTID|SIGCHLD, child_tidptr=0x7f66848d6a10) = 62808
> close(21)                               = 0
> read(20, "", 50000)                     = 0
> close(20)                               = 0
> wait4(62808, 
> ```
> 
> Running ps shows this process as such: `my_username 62808 0.0 0.2 1498376 46436 pts/5 Sl+ 15:07 0:00 /usr/bin/docker start --attach --interactive 807d53fbb165cae550260ed3374e00ce3e468c797a1ee44bfd7df32716358d7b`

seeing exactly same issue, all commands are slow until you let first one execute after hanging for ~5-10s, hangs on same spot, server version 23.0.1
